internal/app/query/query_service.go

```diff
@@ -107,3 +150,108 @@ func getPageSize(cfg config.View) int {
 
 	return pSize
 }
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+type ticketCache struct {
+	listed map[string]*pb.Ticket
+	err    error
+}
+
+func (tc *ticketCache) query(pool *pb.Pool) ([]*pb.Ticket, error) {
+	if tc.err != nil {
+		return nil, tc.err
+	}
+
+	var results []*pb.Ticket
+
+	for _, ticket := range tc.listed {
+		if filter.InPool(ticket, pool) {
+			results = append(results, ticket)
+		}
+	}
+
+	return results, nil
+}
+
+// Does not fail, but may set error which will be returned to clients.
+func (tc *ticketCache) update(store statestore.Service) {
+	previousCount := len(tc.listed)
+
+	currentAll, err := store.GetIndexedIds(context.Background())
+	if err != nil {
+		tc.err = err
+		return
+	}
+
+	deletedCount := 0
+	for id := range tc.listed {
```

Comment:
Would it be better to parallelize these checks?

Reply:
parallelize what?  This requires modifying the ticket map, which can't be done in parallel.

Reply:
Ah sorry for the confusion, I'm not referring to this specific for loop. Would it be more performant if you rewrite this function with the following pattern? 

```
var wg *sync.Waitgroup = &sync.WaitGroup{}
wg.Add(2)

go func() {
  defer wg.Done()
  // get toDelete
}

go func() {
  defer wg.Done()
  // get toFetch
}

wg.Wait()
// delete and fetch
```